src/wallet/wallet.cpp

```diff
@@ -4101,7 +4098,11 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
         LogPrintf("mapAddressBook.size() = %u\n",  walletInstance->mapAddressBook.size());
     }
 
-    return walletInstance;
+    walletInstance->m_last_block_processed = chainActive.Tip();
+
+    CWallet* wallet_ptr = walletInstance.release();
+    RegisterValidationInterface(wallet_ptr);
+    return wallet_ptr;
```

Comment:
Release on return?

Reply:
Any particular reason? I did it this way so that it would be released before `RegisterValidationInterface` creates a global reference. Basically, if RegisterValidationInterface failed for some reason, a memory leak is probably better than a segfault. I don't feel too strongly though.

Reply:
> RegisterValidationInterface failed for some reason

We would want to shutdown no? Anyway leave as it is then.